### PR TITLE
Protect document creation to avoid collision with http routes

### PR DESCRIPTION
**Pull request by dbengsch**
For instance, if a document is created with the _id , it would conflict with the _create route.
Underscores can't be the first character of an _id.

### CODE REVIEW

File: lib/api/controllers/util/requestAssertions.js [hunk 1]
```diff
@@ -80,6 +80,15 @@ module.exports = {
   /**
    * @param {Request} request
    */
+  assertIdStartsNotUnderscore: function requestAssertHasId (request) {
```
Comment (scottinet) [inline lib/api/controllers/util/requestAssertions.js]: Well... Technically, an undefined or null ID does not start with an underscore.
So I personally would modify this method to return without throwing if the id is not defined, allowing to get rid of all the `if (request.input.resource.id) {` lines embedding this assertion call.

File: lib/api/controllers/util/requestAssertions.js [hunk 2]
```diff
@@ -80,6 +80,15 @@ module.exports = {
...
+    if (request.input.resource._id.indexOf('_') === 0) {
```
Comment (scottinet) [inline lib/api/controllers/util/requestAssertions.js]: I propose a more efficient way to perform this check:

`if (request.input.resource.id.charAt(0) === '_') {`

File: lib/api/controllers/util/requestAssertions.js (hunk 1 above)
Comment (ballinette) [inline lib/api/controllers/util/requestAssertions.js]: and rename `function requestAssertHasId ` (which is copied/pasted from the previous function) by `function requestAssertIdStartsNotUnderscore`

**Comment by dbengsch** [comment]
@scottinet @ballinette done

**Pull request merged**